### PR TITLE
[libx265] Remove register classifier (deprecated in C++17)

### PR DESCRIPTION
**Pull request by daravi**
Specify library name and version:  **lib/1.0**
- [✓] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [✓] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [✓] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [✓] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

This adds a patch to make it possible to compile with C++17 flag with newer versions of clang which would otherwise fail due to deprecation error. I've also submitted a patch to the upstream ([see here](https://bitbucket.org/multicoreware/x265/issues/544/register-storage-class-specifier)).

### CODE REVIEW

File: recipes/libx265/all/conanfile.py [hunk 1]
```diff
@@ -49,6 +49,9 @@ def configure(self):
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("x265_{}".format(self.version), self._source_subfolder)
+        if self.version in self.conan_data["patches"]:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
```
Comment (madebr) [inline recipes/libx265/all/conanfile.py:54]:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```
And also please move this to the build method so the sources are kept pristine.
When you apply the patch in the `source` method, there would be no point in storing the patches with the sources.

Comment (daravi) [reply recipes/libx265/all/conanfile.py]: Done. Thanks.

Comment (daravi) [reply recipes/libx265/all/conanfile.py]: @madebr Sorry I was not familiar with GitHub's "Commit suggestion" feature. I didn't mean to not give you credit for the change.

Comment (madebr) [reply recipes/libx265/all/conanfile.py]: Don't worry, my patch was incomplete as I didn't add it to the build method

**Comment by daravi** [comment]
I've requested to be added to the EAP. Does the bot automatically remove the No Beta User flag from this PR or would I need help from a moderator?

**Comment by madebr** [comment]
@daravi 
This is a manual process.
I've asked on #conan @ cpplang.slack.com when new people will be admitted.

**Pull request merged**